### PR TITLE
Test/while

### PR DESCRIPTION
**Pull request by bertschi**
New test for issue #796.
Issue has been fixed in master and test is currently passing.

### CODE REVIEW

File: test/gradcheck.jl [hunk 1]
```diff
@@ -2045,3 +2045,19 @@ end
   h(x) = sum(abs2, g(x))
   @test gradient(h, x)[1] isa typeof(x)
 end
+
+@testset "Zygote issue 796" begin
```
Comment (ToucheSir) [inline test/gradcheck.jl:2049]:
```suggestion
@testset "Zygote #796" begin
```

**Review by ToucheSir — APPROVED** [review]
LGTM, one small style change for your consideration.

File: test/gradcheck.jl [hunk 2]
```diff
@@ -2045,3 +2045,19 @@ end
...
+        while abs(x - y) > 1e-6
+            y, x = (x + y) / 2, y
+        end
+        return y
+    end
+
+    @test gradcheck(foo ∘ first, [0.0])
+    @test gradcheck(foo ∘ first, [2.0])
+    @test gradcheck(foo ∘ first, [-1e-5])
+    @test gradient(foo, 1024.0)[1] ≈ 2//3
```
Comment (MariusDrulea) [inline test/gradcheck.jl:2062]: I would also add several tests such that Zygote matches ForwardDiff and ReverseDiff results; of course, this assumes both ADs give correct results.

Comment (bertschi) [reply test/gradcheck.jl:2062]: Did not include those as all other test sets in this file use finite differences.

**Pull request merged**